### PR TITLE
Add bearer token to headers

### PR DESCRIPTION
**Pull request by mokudekiru**
Adds support for attaching an authorization token to http headers on POST requests – useful when sending events to a custom server.

If this would cause any problems if sent to Amplitude's default server url happy to change it to ignore the token in that case.

### CODE REVIEW

File: Amplitude/Amplitude.m [hunk 1]
```diff
@@ -1023,6 +1025,12 @@ - (void)makeEventUploadPostRequest:(NSString*) url events:(NSString*) events num
     [request setValue:@"application/x-www-form-urlencoded" forHTTPHeaderField:@"Content-Type"];
     [request setValue:[NSString stringWithFormat:@"%lu", (unsigned long)[postData length]] forHTTPHeaderField:@"Content-Length"];
 
+    if (_token != nil) {
+        NSString *auth = [NSString stringWithFormat:@"Bearer %@", _token];
+        AMPLITUDE_LOG(@"Attaching bearer");
```
Comment (haoliu-amp) [inline Amplitude/Amplitude.m]: Do we wanna include the authToken together with this log?

File: Amplitude/Amplitude.m [hunk 2]
```diff
@@ -1474,6 +1482,17 @@ - (void)setServerUrl:(NSString*) serverUrl
     self->_serverUrl = serverUrl;
 }
 
+- (void)setBearerToken:(NSString *)token
+{
```
Comment (haoliu-amp) [inline Amplitude/Amplitude.m]: move this curly bracket to Ln 1485

File: Amplitude/Amplitude.m (hunk 1 above)
Comment (mokudekiru) [reply Amplitude/Amplitude.m]: Sure let's

**Comment by mokudekiru** [comment]
@haoliu-amp rebased just now to fix merge conflicts that had arisen.

**Comment by haoliu-amp** [comment]
Thank you! It’s merged.

**Pull request merged**